spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/publisher/PubSubPublisherTemplate.java

```diff
@@ -84,16 +84,25 @@ public PubSubPublisherTemplate setMessageConverter(PubSubMessageConverter pubSub
 	 */
 	@Override
 	public <T> ListenableFuture<String> publish(String topic, T payload, Map<String, String> headers) {
+		Assert.hasText(topic, "The topic can't be null or empty.");
+		Assert.notNull(payload, "The payload can't be null.");
+
 		return publish(topic, this.pubSubMessageConverter.toPubSubMessage(payload, headers));
 	}
 
 	@Override
 	public <T> ListenableFuture<String> publish(String topic, T payload) {
+		Assert.hasText(topic, "The topic can't be null or empty.");
+		Assert.notNull(payload, "The payload can't be null.");
```

Comment:
We don't need this here because we delegate to the `publish(String topic, T payload, Map<String, String> headers)`

Reply:
Fixing.